vx_nn/src/tensor_table_lookup.cpp

```
@@ -0,0 +1,321 @@
+{
+    vx_enum lut_type, input_type, output_type;
+    vx_size input_ndims = 0, output_ndims = 0;
+    vx_size input_dims[4], output_dims[4];
+    vx_uint8 input_fixedpt_pos = 0, output_fixedpt_pos = 0;
+
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_NUMBER_OF_DIMS, &input_ndims, sizeof(input_ndims)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DATA_TYPE, &input_type, sizeof(input_type)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, input_dims, sizeof(input_dims)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_FIXED_POINT_POSITION, &input_fixedpt_pos, sizeof(input_fixedpt_pos)));
+    if((input_type != VX_TYPE_UINT8) && (input_type != VX_TYPE_INT16)) return VX_ERROR_INVALID_TYPE;
+
+    ERROR_CHECK_STATUS(vxQueryLUT((vx_lut)parameters[1], VX_LUT_TYPE, &lut_type, sizeof(lut_type)));
+
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_NUMBER_OF_DIMS, &output_ndims, sizeof(output_ndims)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_DATA_TYPE, &output_type, sizeof(output_type)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_DIMS, output_dims, sizeof(output_dims)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_FIXED_POINT_POSITION, &output_fixedpt_pos, sizeof(output_fixedpt_pos)));
+
+    if(input_ndims != output_ndims) return VX_ERROR_INVALID_DIMENSION;
+    if((input_ndims >= 1) && (input_dims[0] != input_dims[0])) return VX_ERROR_INVALID_DIMENSION;
+    if((input_ndims >= 2) && (input_dims[1] != input_dims[1])) return VX_ERROR_INVALID_DIMENSION;
+    if((input_ndims >= 3) && (input_dims[2] != input_dims[2])) return VX_ERROR_INVALID_DIMENSION;
+    if((input_ndims >= 4) && (input_dims[3] != input_dims[3])) return VX_ERROR_INVALID_DIMENSION;
+    if(lut_type != output_type) return VX_ERROR_INVALID_TYPE;
+
```

There is a disconnect between the validate and codegen functions when input_type is VX_TYPE_INT16 and output_type is VX_TYPE_UINT8. This function should return an error for this case.